File: core/src/com/alta_v2/rendering/ScreenFactory.java
package com.alta_v2.rendering;

import com.alta_v2.rendering.menuScreen.MenuScreen;
import com.alta_v2.rendering.tiledMapScreen.TiledMapMetadata;
import com.alta_v2.rendering.tiledMapScreen.TiledMapScreen;

/**
 * Provides the factory for creating screens.
 */
public interface ScreenFactory {

    /**
     * Creates the screen that represents the game menu.
     *
     * @return created {@link MenuScreen} instance.
     */
    MenuScreen createMenuScreen();

    /**
     * Creates the screen that represents a game tiled map.
     *
     * @return created {@link TiledMapScreen} instance.
     */
    TiledMapScreen createTiledMapScreen(TiledMapMetadata metadata);

}

File: core/src/com/alta_v2/rendering/ScreenFactory.kt
package com.alta_v2.rendering

import com.alta_v2.rendering.menuScreen.MenuScreen
import com.alta_v2.rendering.tiledMapScreen.TiledMapMetadata
import com.alta_v2.rendering.tiledMapScreen.TiledMapScreen

/**
 * Provides the factory for creating screens.
 */
interface ScreenFactory {
    /**
     * Creates the screen that represents the game menu.
     *
     * @return created [MenuScreen] instance.
     */
    fun createMenuScreen(): MenuScreen

    /**
     * Creates the screen that represents a game tiled map.
     *
     * @return created [TiledMapScreen] instance.
     */
    fun createTiledMapScreen(metadata: TiledMapMetadata): TiledMapScreen
}